src/controller/productController.js

```diff
@@ -0,0 +1,153 @@
+const knex = require("../middleware/database");
+const response = require('../utilities/response');
+const productRepo = require("../repo/product");
+const ratingRepo = require("../repo/rating");
+const { Container } = require("winston");
+const { Console } = require("winston/lib/winston/transports");
+
+const create = async (req, res) => {
+  const { name, description, brand, price, category, image, cart } = req.body;
+  try {
+    const newProduct = await productRepo.create({ name, description, brand, price, category, image, cart })
+    response.success(res, newProduct);
+  } catch (error) {
+    response.error(res, error, 500);
+  }
+};
+
+const updateprod = async (req, res) => {
+  const { id } = req.params;
+  const { name, description, price, category, image, cart } = req.body;
+  try {
+    const result = await productRepo.update({ id, name, description, price, category, image, cart });
+    response.success(res, {
+      message: "Post updated successfully",
+      data: result,
+    });
+  } catch (error) {
+    response.error(res, error, 500);
+  }
+};
+
+const deleteprod = async (req, res) => {
+  const { id } = req.params;
+  try {
+    const deleteCount = await productRepo.remove({ id });
+    if (deleteCount === 0) {
+      return res.status(401).json({ message: "Unauthorized to delete others' product" });
+    } else {
+      response.success(res, {
+        message: "Post product successfully",
+      });
+    }
+  } catch (error) {
+    response.error(res, error, 500);
+  }
+};
+
+const allprod = async (req, res) => {
+  try {
+    const product = await productRepo.allprod();
+    if (product.length === 0) {
+      return res.status(404).json({ message: "No product found" });
+    } else {
+      response.success(res, product);
+    }
+  } catch (error) {
+    response.error(res, error, 500);
+  }
+};
+
+const filterProduct = async (req, res) => {
+  try {
+    const { category, priceRange, brand } = req.query;
+    if (category) {
+      query = await knex('product').select('*').where('category', category);
+    }
+    if (priceRange) {
+      const [minPrice, maxPrice] = priceRange.split('-');
+      query = await knex('product').select('*').whereBetween('price', [minPrice, maxPrice]);
+    }
+    if (brand) {
+      query = await knex('product').select('*').where({ brand });
+    }
+    response.success(res, query);
+  } catch (error) {
+    response.error(res, error, 500);
+  }
+};
+
+const searchProduct = async (req, res) => {
+  try {
+    const { name, category } = req.query;
+    const searchResults = await knex('product').select('*').where('name', name).orWhere('category', category);
+    response.success(res, searchResults);
```

Comment:
```
response.success(res, searchResults);
```
Does this work?
Did you test with API calls ( postman )?

because I  can't see `response` variable ( able to see the `res` variable )

Reply:
yes, I rechecked it today. It's working. 
![image](https://github.com/poulamib1602/e-commerce-postgresql/assets/148246809/64bb0132-82bc-4916-84d7-aaa78718e5f4)

Reply:
Sometimessome times you use 
```
response.success(res, searchResults);
```
and sometimes uses 

```
res.send
```

that's why confused